client/scala/src/main/scala/ai/verta/blobs/dataset/PathBlob.scala

```diff
@@ -0,0 +1,122 @@
+package ai.verta.blobs.dataset
+
+import ai.verta.swagger._public.modeldb.versioning.model._
+
+import java.security.{MessageDigest, DigestInputStream}
+import java.io.{File, FileInputStream}
+
+import scala.collection.mutable.HashMap
+import scala.util.{Failure, Success, Try}
+
+/** Captures metadata about files
+ *  @param paths list of filepaths or directory paths
+ */
+ case class PathBlob(private val paths: List[String]) extends Dataset {
+  private val BufferSize = 8192
+  paths.map(expanduser _).map((path: String) => processPath(new File(path)))
```

Comment:
Ok I can make it more functional by mapping from a list of item to list of (key -> value) before converting it to map, but IMO it still looks a bit ugly because I'm using HashMap. I think it's a bit more efficient than immutable map right? But if it's not worth it then I can just call toMap on the list of pair and call it a day. Resolved for now.